qf/types/requests.proto

```diff
@@ -0,0 +1,172 @@
+syntax = "proto3";
+package types;
+option go_package   = "github.com/quickfeed/quickfeed/qf/types";
+option swift_prefix = "";
+
+import "qf/types/types.proto";
+
+// Message types that are used as requests and responses in API calls.
+
+message SubmissionLink {
+    types.Assignment assignment = 1;
+    types.Submission submission = 2;
+}
+
+message EnrollmentLink {
+    types.Enrollment enrollment         = 2;  // must have course and user/group preloaded
+    repeated SubmissionLink submissions = 3;
+}
+
+message CourseSubmissions {
+    types.Course course           = 1;  // preloaded assignments
+    repeated EnrollmentLink links = 2;
+}
+
+message ReviewRequest {
+    uint64 courseID     = 1;
+    types.Review review = 2;
+}
+
+message CourseRequest {
+    uint64 courseID = 1;
+}
+
+message UserRequest {
+    uint64 userID = 1;
+}
+
+message GetGroupRequest {
+    uint64 groupID = 1;
+}
+
+message GroupRequest {
+    uint64 userID   = 1;
+    uint64 groupID  = 2;
+    uint64 courseID = 3;
+}
+
+message Provider {
```

Comment:
This actually don't seem to be used at all and can probably be removed. There is only a `IsValid()` method on this, but otherwise it doesn't seem like it is being used anymore.

Reply:
Right. Removed it.